elasticsearch/mapping_sirene_index.py

```diff
@@ -94,79 +152,42 @@ class ElasticsearchSireneIndex(Document):
-    adresse_etablissement = Text(analyzer=annuaire_analyzer)
     categorie_entreprise = Text()
-    cedex = Keyword()
-    code_pays_etranger = Text()
-    code_postal = Keyword()
     colter_code = Keyword()
     colter_code_insee = Keyword()
     colter_elus = Nested(ElasticsearchEluIndex)
     colter_niveau = Keyword()
-    commune = Keyword()
-    complement_adresse = Text()
-    concat_enseigne_adresse = Text(analyzer=annuaire_analyzer)
-    concat_nom_adr_siren = Text(
-        analyzer=annuaire_analyzer, fields={"keyword": Keyword()}
-    )
-    coordonnees = GeoPoint()
-    date_creation_siege = Date()
     date_creation_unite_legale = Date()
-    date_debut_activite_siege = Date()
-    date_mise_a_jour = Date()
-    departement = Keyword()
+    date_mise_a_jour_unite_legale = Date()
     dirigeants_pp = Nested(ElasticsearchDirigeantPPIndex)
     dirigeants_pm = Nested(ElasticsearchDirigeantPMIndex)
-    distribution_speciale = Text()
     economie_sociale_solidaire_unite_legale = Keyword()
-    enseigne = Text(analyzer=annuaire_analyzer)
-    etat_administratif_unite_legale = Keyword()
-    etat_administratif_siege = Keyword()
-    geo_adresse = Text(analyzer=annuaire_analyzer)
-    geo_id = Keyword()
-    identifiant_association_unite_legale = Keyword()
-    indice_repetition = Text()
     est_entrepreneur_individuel = Boolean()
     est_entrepreneur_spectacle = Boolean()
-    est_siege = Boolean()
-    latitude = Text()
-    libelle_cedex = Text()
-    libelle_commune = Text()
-    libelle_commune_etranger = Text()
-    libelle_pays_etranger = Text()
-    libelle_voie = Text()
-    liste_adresses = Text(analyzer=annuaire_analyzer)
-    liste_idcc = Text()
+    etablissements = Nested(ElasticsearchEtablissementIndex)
+    etat_administratif_unite_legale = Keyword()
+    identifiant_association_unite_legale = Keyword()
     liste_dirigeants = Text(analyzer=annuaire_analyzer)
     liste_elus = Text(analyzer=annuaire_analyzer)
-    liste_enseignes = Text(analyzer=annuaire_analyzer)
-    liste_finess = Text()
-    liste_rge = Text()
-    liste_uai = Text()
-    longitude = Text()
     nature_juridique_unite_legale = Integer()
     nom = Text(analyzer=annuaire_analyzer)
     nom_complet = Text(analyzer=annuaire_analyzer, fields={"keyword": Keyword()})
     nom_raison_sociale = Text()
     nombre_etablissements = Integer()  # NaN can't be stored in an integer array
     nombre_etablissements_ouverts = Integer()
-    numero_voie = Text()
     prenom = Text(analyzer=annuaire_analyzer)
     section_activite_principale = Keyword()
     sigle = Keyword()
     siren = Keyword(required=True)
-    siret_siege = Keyword(required=True)
-    type_voie = Text()
-    tranche_effectif_salarie_siege = Keyword()
+    siret_siege = Keyword()
     tranche_effectif_salarie_unite_legale = Keyword()
 
     class Index:
         name = f"siren-{NEXT_COLOR}"
         settings = {
-            "number_of_shards": 1,
+            "number_of_shards": 4,
```

Comment:
this could maybe be a variable ?

Reply:
Yes !

Reply:
I added it in the same file for now. It's complicated to add it to the class through a parameter.